tests/t3_inode_cache.py

```diff
@@ -106,3 +106,24 @@ def test_get(self):
             self.cache.create_inode(**attrs)
 
         self.assertRaises(KeyError, self.cache.__getitem__, inode.id)
+
+    def test_filetimes(self):
```

Comment:
Thanks for providing this test, it helps a lot to understand the problem!

Timestamps above 2^63 are in the year 2262. Is there a use-case for a file having such a time-stamp? My thinking is that perhaps a better solution would be to return a -EINVAL to userspace. Looking at the original bug report, it seems these timestamps were the result of an application bug.

The advantage of that would be that the database format is easier to understand and that we do not need to bump the filesystem revision. Furthermore, the solution above still fails for timestamps above 2*2^63 (i.e., after year 2554+).